src/probnum/problems/_problems.py

```diff
@@ -1 +1,204 @@
-"""Definitions of problems solved by probabilistic numerical methods."""
+"""Definitions of problems currently solved by probabilistic numerical methods.
+
+If a problem is not listed here, we have not implemented a probabilistic
+numerical method for it yet.
+"""
+
+import dataclasses
+import typing
+
+import numpy as np
+import scipy.sparse
+
+import probnum.filtsmooth as pnfs
+import probnum.linops as pnlo
+import probnum.random_variables as pnrv
+import probnum.type as pntp
+
+
+@dataclasses.dataclass
+class RegressionProblem:
+    r"""Regression problem.
+
+    Fit a stochastic process to data, given a likelihood (realised by a :obj:`DiscreteGaussian` transition).
+    Solved by Kalman filtering and smoothing in :mod:`probnum.filtsmooth`.
```

Comment:
Isn't RegressionProblem too broad here? There is at least a Markovian assumption if this is solved by Kalman filtering, right? I would recommend making either the docstring more general or changing the name.

Reply:
No, in fact it is not (at least that's what I hope). This should include any GP regression problems. I see that it might not be clear, because the likelihood is defined by something in `statespace`, but that is just a sub-package artefact (and one of the reasons I want to pull up statespace and rename it). In 1d, with markovian structure, Kalman filters solve it. The problem definition is more general than that though.